src/pages/resource/ResourceUsage.tsx

```diff
@@ -53,8 +56,29 @@ const ResourceUsage: FC<ResourceUsageProps> = ({ resource }) => {
   const [citationPreTitle, setCitationPreTitle] = useState('');
   const [citationTitle, setCitationTitle] = useState('');
   const [citationPostTitle, setCitationPostTitle] = useState('');
+  const [showCopiedLinkToClipboardInformation, setShowCopiedLinkToClipboardInformation] = useState(false);
   const mountedRef = useRef(true);
 
+  const generatePreferredURL = (): string => {
+    if (resource.features.dlr_identifier_doi) {
+      return resource.features.dlr_identifier_doi;
+    } else if (resource.features.dlr_identifier_handle) {
+      return resource.features.dlr_identifier_handle;
+    } else {
+      return `${window.location.origin}/resource/${resource.identifier}`;
+    }
+  };
+
+  const generateLinkSharingText = (): string => {
```

Comment:
noen ganger kan det være skummelt å ha sånne funksjoner i en komponent siden de vil opprettes på nytt for hver render. Hvis komponenten vil re-rendres ofte kan det bli litt lagging. Alternativet er å enten legge det utenfor komponenten, eller å wrappe det i en `useCallback` slik at den bare opprettes når det har skjedd relevante endringer

Reply:
Fikset